Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/conceptcollection/ConceptCollectionDAO.java

```diff
@@ -42,376 +39,347 @@
+            query.setParameter("collectionname", collectionname);
+            List<ConceptCollectionDTO> conceptcollectionsDTOList = query.list();
+
+            if (!(conceptcollectionsDTOList != null && conceptcollectionsDTOList.size() > 0)) {
+                errMsg = "collection id is invalid.";
+            }
+        } catch (Exception e) {
+            logger.info("getCollaboratedConceptsofUser method :" + e.getMessage());
+            throw new QuadrigaStorageException(e);
+        }
+        return errMsg;
+    }
+
+    /**
+     * This method deletes the given items associated with the specified concept
+     * collection
+     * 
+     * @param :
+     *            id - concept collection associated item ids.
+     * @param :
+     *            collectionid - concept collection id
+     * @param :
+     *            username - logged in user
+     */
+    @SuppressWarnings("unchecked")
+    @Override
+    public String deleteItems(String id, String collectionId, String username) throws QuadrigaStorageException {
+        String errMsg = null;
+        if (id == null || collectionId == null || username == null || id.isEmpty() || collectionId.isEmpty()
+                || username.isEmpty()) {
+            logger.error("deleteItems: input argument IConceptCollection is null");
+            return null;
+        }
+        try {
+            Query query = sessionFactory.getCurrentSession().createQuery(
+                    "from ConceptCollectionItemsDTO ccItems where ccItems.conceptCollectionItemsDTOPK.conceptcollectionid =:collectionId and ccItems.conceptCollectionItemsDTOPK.concept =:id");
+            query.setParameter("id", id);
+            query.setParameter("collectionId", collectionId);
+            List<ConceptCollectionItemsDTO> ccItemsDTOList = query.list();
+
+            if (ccItemsDTOList != null && ccItemsDTOList.size() > 0) {
+                sessionFactory.getCurrentSession().delete(ccItemsDTOList.get(0));
+            } else {
+                errMsg = "collection id is invalid.";
+            }
+            Query query1 = sessionFactory.getCurrentSession().createQuery(
+                    "from ConceptCollectionItemsDTO ccItems where ccItems.conceptCollectionItemsDTOPK.concept =:id");
+            query1.setParameter("id", id);
+            List<ConceptCollectionItemsDTO> ccItemsDTOList1 = query1.list();
+            if (ccItemsDTOList1 == null || ccItemsDTOList1.size() == 0) {
+                Query query2 = sessionFactory.getCurrentSession()
+                        .createQuery("from ConceptsDTO concepts where concepts.item =:id");
+                query2.setParameter("id", id);
+                List<ConceptsDTO> conceptsList = query2.list();
+                if (conceptsList != null && conceptsList.size() > 0) {
+                    sessionFactory.getCurrentSession().delete(conceptsList.get(0));
+                }
+            }
+
+        } catch (Exception e) {

```

Comment:
you might keep it here though, since there are two blocks that might throw an exception. but remove the logger and catch a specific exception.